### PR TITLE
Add Serverless Workflow Specification v0.5 definition models

### PR DESCRIPTION
**Pull request by Neuroglia**
**Many thanks for submitting your Pull Request :heart:!**

**What this PR does / why we need it**:

**Special notes for reviewers**:
Adds the Serverless Workflow Specification v0.5 definition models

**Additional information (if needed):**
Coming soon:
- YAML/JSON Parsers
- Validation

### CODE REVIEW

**Comment by tsurdilo** [comment]
@Neuroglia thanks! this is really cool. 
Just one question, is there a way to update the README.md so that .NET noobs like myself can try it out? Thanks!

File: src/ServerlessWorkflow.Sdk/ActionExecutionMode.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+﻿/*
+ * Copyright 2020-Present The Serverless Workflow Specification Authors
```
Comment (ricardozanini) [inline src/ServerlessWorkflow.Sdk/ActionExecutionMode.cs:2]:
```suggestion
 * Copyright 2021-Present The Serverless Workflow Specification Authors
```

Sorry Charles, could you please change the headers?

**Review by ricardozanini — COMMENTED** [review]
Seems nice! So for this PR we are bringing the data structure as we have on 0.6, correct?

Just a comment about the date in the headers. Sorry to nitpick. :disappointed:

**Comment by ricardozanini** [comment]
@jplane can you also please take a look?

File: src/ServerlessWorkflow.Sdk/ActionExecutionMode.cs (hunk 1 above)
Comment (Neuroglia) [reply src/ServerlessWorkflow.Sdk/ActionExecutionMode.cs:2]: Yes, I will fix that tomorrow morning, no problem ;)

**Comment by Neuroglia** [comment]
@tsurdilo I will update the Readme tomorrow morning, along with the addition of the parsers ;)

File: src/ServerlessWorkflow.Sdk/ActionExecutionMode.cs (hunk 1 above)
Comment (Neuroglia) [reply src/ServerlessWorkflow.Sdk/ActionExecutionMode.cs:2]: I'm not sure what has changed in 0.6 from latest 0.5x, but I will make sure to update whatever needs to be.

Comment (Neuroglia) [reply src/ServerlessWorkflow.Sdk/ActionExecutionMode.cs:2]: @ricardozanini I just reviewed the 0.6 and it seems my code is up-to-date. I'll change the version to 0.6

**Comment by Neuroglia** [comment]
Sorry for the delay guys, I've got a lot on my plate and I thought it would be a great idea to add fluent builders before checking in, which is taking far more time than I originally thought. I will commit my changes ASAP and will keep you posted ;)

**Comment by tsurdilo** [comment]
@Neuroglia np at all, take your time and make sure you get rest too! Looked at the updated README - wow, really cool !

**Review by ricardozanini — APPROVED** [review]
Pretty nice man! Many thanks

Just one thing: I believe you have to sign your commit:

`git commit --amend --signoff`

That's why the CI is failing. :thinking:

**Comment by tsurdilo** [comment]
@Neuroglia @ricardozanini so is this ok to be merged?

**Comment by Neuroglia** [comment]
@tsurdilo @ricardozanini Well I still need to checkin the reader and writer tomorrow morning but then yes, it could be merged. I will make sure to sign-off too!

**Comment by Neuroglia** [comment]
@tsurdilo Working right now on signing the commit. Then IMHO opining, everything's ready to go.
Only part missing so far is schema and integrity validation, but they'll be coming very soon.

**Comment by ricardozanini** [comment]
@Neuroglia all your commits must be signed. You can squash all of them in one, sign, and force push.

**Comment by Neuroglia** [comment]
@ricardozanini I a Git noob, sorry guys! How can I sign them off all at once?

**Comment by tsurdilo** [comment]
@Neuroglia don't worry about it :) We can just merge.
For next ones try git commit -s -m "...."
Thank you!

**Pull request merged**